src/library/parserpls.cpp

```diff
@@ -18,27 +6,40 @@
 #include <QUrl>
 #include <QtDebug>
 
-#include "moc_parserpls.cpp"
+#include "library/parser.h"
 
-/**
-   ToDo:
-    - parse ALL information from the pls file if available ,
-          not only the filepath;
- **/
+namespace {
 
-ParserPls::ParserPls() : Parser() {
-}
+QString getFilePath(QTextStream* stream) {
```

Comment:
why the ptr instead of a const-ref?

Reply:
It modifies the stream (moves the seek position) so I think a ptr is semantically appropriate.

Reply:
Ah, right, so the text stream must be mutable, but that still does not explain why were passing a pointer instead of a (mutable) reference. AFAIK they are semantically the same, so is the pointer just used because it was there in the first place?

Reply:
Keep in mind that this is not a request for changes, I'm must asking because I'd like improve my C++ knowledge.

Reply:
I will change "stream" to "pStream" ...

From the C++ rules we may pass here a non const reference: 
`QString getFilePath(QTextStream& stream);` But it is a matter of code style.  

Than we see: 
`stream.readLine();`
and not 
`pStream->readLine();`

The later makes clear, that we do not have a local copy and all work with the pointed object will have an outside effect. 

In case of `QString getFilePath(cosnt QTextStream& stream);` it does not matter if it is a copy by value or a reference, It has never an outside effect, because of the constantnes 
  
This a fundamental difference between C++ and Java or C#, where objects are always a references and where thy are no pointers. 

This was also part of the Google style. However they have recently dropped the strict rule.

See also: 
https://github.com/mixxxdj/mixxx/wiki/Coding-Guidelines#non-const-references

Reply:
Thanks for the explanation. Stylewise, when passing a pointer, while passing `nullptr` would not make sense, it is still permitted. Wouldn't it make sense to use a mutable reference instead since those can't be null?
Moreover, in this particular case, we don't treat the incoming QTextStream as an output variable, but rather consume the textstream and expect it to not be used in the caller afterwards, correct? Because then the parameter rather seems like it should semantically be a `QTextStream&&`

Reply:
> Wouldn't it make sense to use a mutable reference instead since those can't be null?

That is a valid argument for the other code style. Both styles are valid, we prefere the reliable value vs. pointer syntax style. By the way: in C++ a mutable reference can be null. It is only by convention never null. Pointers and references are exactly the same if you look at the produced assembler code.

Reply:
> That is a valid argument for the other code style. Both styles are valid, we prefere the reliable value vs. pointer syntax style.

Thanks for the clarification. I guess I'd prefer the the other style but I can adopt the mixxx style as well. 

>  By the way: in C++ a mutable reference can be null. It is only by convention never null. Pointers and references are exactly the same if you look at the produced assembler code.

I know, but I thought the compiler does not allow references to be set to null (at compile time, there aren't any runtime checks unless you use something like `gsl::not_null`).